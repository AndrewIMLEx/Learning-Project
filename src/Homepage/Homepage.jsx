import React from "react";

class Homepage extends React.Component{
    render() {
      return (
      <>
      <div className="card">
        <h1>Guess The Number</h1> 
        <h2>Registration Page</h2>
        </div>
      </>
      )
    }
}

export default Homepage
