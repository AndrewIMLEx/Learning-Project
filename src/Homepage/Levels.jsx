import React, { Component} from "react"

class Levels extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            levelSelected: "LEVEL",             
            input: "",
            tooHightooLow: "",
            standardNum: Math.floor(Math.random() *10) + 1 + "",
            difficultNum: Math.floor(Math.random() *100) + 1 + "",
        };
        this.standard = this.standard.bind(this);
        this.difficult = this.difficult.bind(this);
        this.guessNumber = this.guessNumber.bind(this);
        this.clickedbuttonGuess = this.clickedbuttonGuess.bind(this);
        this.Initial = this.Initial.bind(this)
    }
    Initial(){
        this.setState({
            levelSelected: "LEVEL",             
            input: "",
            tooHightooLow: "",
            guessNumber: "",
            standardNum : Math.floor(Math.random() *10) + 1 + "",
            difficultNum: Math.floor(Math.random() * 100 + 1) + "",         
    })
    }
    standard(e){
        e.preventDefault();
        this.setState({levelSelected:'STANDARD'});
    }
    difficult(e){
        e.preventDefault();
        this.setState({levelSelected:'DIFFICULT'});
    }
    componentDidMount(){
        if(this.standard){
            this.setState({standardNum : Math.floor(Math.random() *10) + 1 + ""})
        }
        else if(this.difficult){
            this.setState({difficultNum: Math.floor(Math.random() *100) + 1 + ""})
        }
    }
    componentDidUpdate(){
      
    }
    guessNumber(e){
        e.preventDefault();
        this.setState({ input:e.target.value})
    }
clickedbuttonGuess(e){
    e.preventDefault(); 
    let startNewStandardGame = this.state.standardNum;
    let startNewDifficultGame = this.state.difficultNum;
    let guessedValue = this.guessNumber;  

    console.log(this.state.levelSelected);
    console.log(this.state.input);

        if(guessedValue == startNewStandardGame){
            this.setState({tooHightooLow:'Good Job!Correct'});
        }else if(guessedValue < startNewStandardGame){
            this.setState({tooHightooLow:'Wrong!Too Low'});
        }else if(guessedValue  > startNewStandardGame){
            this.setState({tooHightooLow:'Wrong!Too high'});
        }
        if(guessedValue  == startNewDifficultGame){
            this.setState({tooHightooLow:'Good Job!Correct'});
        }else if(guessedValue  > startNewDifficultGame){
            this.setState({tooHightooLow:'Wrong!too High'});
        }else if(guessedValue  < startNewDifficultGame){
            this.setState({tooHightooLow:'Wrong!Too Low'});
        }
        else {
            this.setState({tooHightooLow:'PLEASE ENTER A NUMBER'});
        }
    }
render(){
    return(
        <>
        <div className="levelselect">
        <h5 style={{color: "white",fontSize:"17px",marginLeft:'5em'}}>Select Difficulty Level</h5>
        <button id="standard" value={this.state.standard}  onClick ={this.standard}>Standard</button>
        <button id="difficult" value={this.state.difficult} onClick ={this.difficult}>Difficult</button>
        </div>
        <div className="levelCard">
        <h3 style={{color: "black",fontSize: '20px',marginLeft: '6em'}}>Selected Level</h3>
        &nbsp;
        <p style={{color: "red",fontSize: 20,marginLeft:'6.9em'}}>{ this.state.levelSelected }</p>
        &nbsp;
        <form id = "LevelForm">
            <input style={{float:'center'}} value={this.state.input}  id = "Levelsinput" placeholder="Enter Number Here" type="number" onChange = {this.guessNumber} ref={this.inputFocus}/>
        </form>
        &nbsp;&nbsp;
        <p style={{color: "red",marginLeft:'6.5em'}}>{this.state.tooHightooLow}</p>
        &nbsp;&nbsp;
        <button onClick={this.clickedbuttonGuess}  id="guess" type = "submit">Guess</button>
        &nbsp;
        <div className="playButtons">
        <button id="play"  onClick={this.Initial} type = "submit">Play Again</button>
        <button id="login" type = "submit">Log In</button>
        </div>
        </div>
        </>
    )
}
}
export default Levels
