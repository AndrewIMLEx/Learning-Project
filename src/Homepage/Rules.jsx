import React from "react";

class Rules extends React.Component{
    render() {
      return (
      <>
      <div className="rules">
        <h3>How to Play</h3>
        <ol>
            <li>Choose Difficulty Level-Easy(1-10) or Hard(1-100).</li>
            <li>The Computer will generate a random number between 1-10 for standard and 1-100 for Difficult.</li>
            <li>You only have 3 chances to guess a number and beat the Computer</li>
            <li>Don't worry the computer will notify you if your guess is higher or lower than the random generated number</li>
        </ol>
        </div>
      </>
      )
    }
}

export default Rules
