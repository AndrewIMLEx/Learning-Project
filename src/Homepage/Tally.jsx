import React from "react"

class Tally extends React.Component{
    render(){
        return(
            <>
            <div className="tallycard">
            <h4 style={{marginLeft:"2.8em",paddingBottom:"2em"}}>Game Tally</h4>
            <form id = "LevelForm">
            <span><label id = "won-label" htmlFor="won">Games Won</label>
            <label id = "Tally"/></span>
            &nbsp;&nbsp;
            <span><label id = "won-label" htmlFor="lost">Games Lost</label>
            &nbsp;<label id = "Tally"/></span>
             </form>
             <form id = "Gamesplayed">
             <label id ="played" htmlFor="gamesplayed" style={{padding:"-4em", marginLeft:"3.4em",display: "inline"}}>Total Games Played</label>
            <label id ="plays"/>
            </form>
            </div>
            </>
        )
    }
}
export default Tally
