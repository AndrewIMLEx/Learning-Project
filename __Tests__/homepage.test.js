import React from 'react';
import Homepage from '../src/Homepage/Homepage';
import renderer from 'react-test-renderer';
describe("Homepage page",() =>{
    it('renders the landing page', () => {
      const tree = renderer.create(<Homepage />).toJSON();
      expect(tree).toMatchSnapshot();
    })
  })
  
