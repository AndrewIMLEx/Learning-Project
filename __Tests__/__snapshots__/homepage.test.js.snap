// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage page renders the landing page 1`] = `
<div>
  Hello World! Its me coming 
</div>
`;
